java-checks/src/main/java/org/sonar/java/checks/CognitiveComplexityMethodCheck.java

```diff
@@ -64,7 +70,15 @@ public void setMax(int max) {
     this.max = max;
   }
 
-  private static boolean isExcluded(MethodTree methodTree) {
-    return MethodTreeUtils.isEqualsMethod(methodTree) || MethodTreeUtils.isHashCodeMethod(methodTree);
+  public void setIgnoredMethods(String ignoredMethods) {
+    this.ignoredMethods = ignoredMethods;
+  }
+
+  private boolean isExcluded(MethodTree methodTree) {
+    return MethodTreeUtils.isEqualsMethod(methodTree) ||
+      MethodTreeUtils.isHashCodeMethod(methodTree) ||
+      Arrays.stream(ignoredMethods.split(";"))
+        .map(ignoredMethodName -> MethodTreeUtils.isNamed(methodTree, ignoredMethodName))
+        .reduce(Boolean.FALSE, (base, acc) -> base || acc);
```

Comment:
This should not be done every time the `isExcluded` method is called. Instead, a set `ignored` of ignored methods should be created from the `ignoredMethods` parameter only once, and `isExcluded` should then call `ignored.contains(methodName)`.